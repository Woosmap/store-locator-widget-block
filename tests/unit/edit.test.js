import {render, cleanup} from "@testing-library/react";
import StoreLocatorBlockEdit from '../../src/edit';

global.ResizeObserver = jest.fn().mockImplementation(() => ({
	observe: jest.fn(),
	unobserve: jest.fn(),
	disconnect: jest.fn(),
}))
describe("StoreLocatorBlockEdit component", () => {
	afterEach(cleanup);

	it("matches snapshot", () => {
		const attributes = {
			height: 300,
			woosmapView: {
				initialCenter: {lat: 50, lng: 0},
				initialZoom: 13,
			},
			theme: {primary_color: '#000000'},
			apiKey: 'fake-apikey'
		};

		const {asFragment} = render(
			<StoreLocatorBlockEdit
				{...{
					attributes,
					setAttributes: jest.fn(),
					clientId: "random-id",
					isSelected: false,
				}}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});

});
