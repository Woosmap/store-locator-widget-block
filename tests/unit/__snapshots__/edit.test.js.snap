// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocatorBlockEdit component matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Block: undefined"
    class="block-editor-block-list__block wp-block"
    id="block-undefined"
    role="document"
    tabindex="0"
  >
    <div
      class="components-placeholder"
    >
      <div
        aria-hidden="true"
        style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; pointer-events: none; opacity: 0; overflow: hidden; z-index: -1;"
      />
      <div
        class="components-placeholder__label"
      >
        Store Locator Widget Configuration
      </div>
      <div
        class="components-placeholder__fieldset"
      >
        <div
          style="margin-bottom: 1em;"
        >
          Before updating the configuration, please ensure that your JSON is valid and that it includes all necessary fields for the widget to function correctly.
        </div>
        <button
          class="components-button is-primary"
          style="margin-top: 10px;"
          type="button"
        >
          Validate Configuration
        </button>
        <button
          class="components-button is-secondary"
          style="margin-top: 10px; margin-left: 10px;"
          type="button"
        >
          Copy Configuration
        </button>
        <pre
          aria-label="Conf"
          aria-multiline="true"
          aria-readonly="false"
          class="block-editor-rich-text__editable rich-text"
          contenteditable="true"
          role="textbox"
          style="white-space: pre-wrap; min-width: 1px;"
        >
          {
          <br
            data-rich-text-line-break="true"
          />
            "woosmapView": {
          <br
            data-rich-text-line-break="true"
          />
              "initialCenter": {
          <br
            data-rich-text-line-break="true"
          />
                "lat": 50,
          <br
            data-rich-text-line-break="true"
          />
                "lng": 0
          <br
            data-rich-text-line-break="true"
          />
              },
          <br
            data-rich-text-line-break="true"
          />
              "initialZoom": 13
          <br
            data-rich-text-line-break="true"
          />
            },
          <br
            data-rich-text-line-break="true"
          />
            "theme": {
          <br
            data-rich-text-line-break="true"
          />
              "primary_color": "#000000"
          <br
            data-rich-text-line-break="true"
          />
            }
          <br
            data-rich-text-line-break="true"
          />
          }
        </pre>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
